src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -58,6 +58,13 @@ def calculate_timeout_seconds(
     return target_timeout
 
 
+@dataclass(frozen=True)
+class PytestRunner(TestRunner):
+    @staticmethod
+    def is_valid_target(adaptor_with_origin: TargetAdaptorWithOrigin) -> bool:
+        return isinstance(adaptor_with_origin, PythonTestsAdaptorWithOrigin)
```

Comment:
This is the most important part of the change. When we have the Target API, we'll be able to say "check that the `Target` has the fields `PythonSources`, `Timeout`, etc". We won't care if it has any additional fields, so long as it has the fields we need. Refer to https://docs.google.com/document/d/1nxPdvuzgCPKhTabhfYBN2tbcr85enSR83OpcvdfWXfY/edit#heading=h.z97og7gj9rvs.

Reply:
Worth moving this comment into the code as a TODO, I think. Or next time.